homeassistant/components/zerproc/light.py

```diff
@@ -0,0 +1,202 @@
+"""Zerproc light platform."""
+from datetime import timedelta
+import logging
+from typing import Callable, List
+
+import pyzerproc
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_HS_COLOR,
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR,
+    Light,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.typing import HomeAssistantType
+import homeassistant.util.color as color_util
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_ZERPROC = SUPPORT_BRIGHTNESS | SUPPORT_COLOR
+
+DISCOVERY_INTERVAL = timedelta(seconds=60)
+
+PARALLEL_UPDATES = 0
+
+
+def connect_lights(lights: List[pyzerproc.Light]) -> List[pyzerproc.Light]:
+    """Attempt to connect to lights, and return the connected lights."""
+    connected = []
+    for light in lights:
+        try:
+            light.connect(auto_reconnect=True)
+            connected.append(light)
+        except pyzerproc.ZerprocException:
+            _LOGGER.debug("Unable to connect to '%s'", light.address, exc_info=True)
+
+    return connected
+
+
+def discover_entities(hass: HomeAssistant) -> List[Entity]:
+    """Attempt to discover new lights."""
+    lights = pyzerproc.discover()
+
+    # Filter out already discovered lights
+    new_lights = [
+        light
+        for light in lights
+        if light.address not in hass.data[DOMAIN]["light_entities"]
+    ]
+
+    entities = []
+    for light in connect_lights(new_lights):
+        # Double-check the light hasn't been added in another thread
+        if light.address not in hass.data[DOMAIN]["light_entities"]:
+            entity = ZerprocLight(light)
+            hass.data[DOMAIN]["light_entities"][light.address] = entity
```

Comment:
We don't need to store the entity. Just make the container a `set` and keep track of the light addresses.